code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -350,14 +348,167 @@
 		/obj/item/reagent_containers/glass/bucket,
 		/obj/item/paint/paint_remover,
 		/obj/item/lightreplacer/cyborg,
-		/obj/item/holosign_creator/janibarrier,
+		/obj/item/holosign_creator,
 		/obj/item/reagent_containers/spray/cyborg_drying)
 	radio_channels = list(RADIO_CHANNEL_SERVICE)
 	emag_modules = list(/obj/item/reagent_containers/spray/cyborg_lube)
 	cyborg_base_icon = "janitor"
 	model_select_icon = "janitor"
 	hat_offset = -5
-	clean_on_move = TRUE
+	/// Weakref to the wash toggle action we own
+	var/datum/weakref/wash_toggle_ref
+
+/obj/item/robot_model/janitor/be_transformed_to(obj/item/robot_model/old_model, forced = FALSE)
+	. = ..()
+	if(!.)
+		return
+	var/datum/action/wash_toggle = new /datum/action/toggle_buffer(loc)
+	wash_toggle.Grant(loc)
+	wash_toggle_ref = WEAKREF(wash_toggle)
+
+/obj/item/robot_model/janitor/Destroy()
+	QDEL_NULL(wash_toggle_ref)
+	return ..()
+
+/datum/action/toggle_buffer
+	name = "Activate Auto-Wash"
+	desc = "Trade speed and water for a clean floor"
+	icon_icon = 'icons/mob/actions/actions_silicon.dmi'
+	button_icon_state = "activate_wash"
+	var/static/datum/callback/allow_buffer_change
+	var/block_buffer_change	= FALSE
+	var/buffer_on = FALSE
+	///The bucket we draw water from
+	var/datum/weakref/bucket_ref
+	///Our looping sound
+	var/datum/looping_sound/wash/wash_audio
+	///Activation cooldown to prevent sound spam
+	COOLDOWN_DECLARE(activation_cooldown)
+
+/datum/action/toggle_buffer/New(Target)
+	if(!allow_buffer_change)
+		allow_buffer_change = CALLBACK(src, .proc/allow_buffer_change)
+	return ..()
+
+/datum/action/toggle_buffer/Destroy()
+	if(buffer_on)
+		toggle_auto_wash()
+	QDEL_NULL(wash_audio)
+	return ..()
+
+/datum/action/toggle_buffer/Grant(mob/M)
+	. = ..()
+	wash_audio = new(owner)
+
+/datum/action/toggle_buffer/IsAvailable()
+	if(!istype(owner, /mob/living/silicon/robot))
+		return FALSE
+	return TRUE
+
+/datum/action/toggle_buffer/Trigger()
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/silicon/robot/robot_owner = owner
+
+	block_buffer_change = DOING_INTERACTION(owner, "auto_wash_toggle")
+	if(block_buffer_change)
+		return FALSE
+
+	var/obj/item/reagent_containers/glass/bucket/our_bucket = locate(/obj/item/reagent_containers/glass/bucket) in robot_owner.model.modules
```

Comment:
It has to be a bucket? I can't use a larger container like a bluespace beaker later?

Reply:
where will you get a bluespace beaker as a janiborg

Reply:
Printing them when available at R&D or whichever department lathe is closest by, is that not a thing here?

Reply:
How are you gonna get a bluespace beaker into your equipment? is the idea to drag it around?

Reply:
sorry for the late reply! i figured the roboticist could swap out whatever reagent container is inside of the janiborg, along with what's in it sorta like deconstructing/upgrading a chemmaster.

Reply:
I don't think that's currently possible, might be ok as a follow up pr. Not something I wanna do now tho